corehq/ex-submodules/dimagi/utils/couch/bulk.py

```diff
@@ -106,9 +104,9 @@ def get_docs(db, keys, **query_params):
     query_params['include_docs'] = True
 
     query_params = {k: json.dumps(v) for k, v in query_params.items()}
-    r = requests.post(url, data=payload,
+    rsession = db._request_session
+    r = rsession.post(url, data=payload,
```

Comment:
It's still not clear to me why this `url` does contain the username/password since `db` is a [`Database`](https://github.com/dimagi/couchdbkit/blob/bde99d120d855e201c069f0746dc3cc11f0886f9/couchdbkit/client.py#L254), which passes it's `uri` (which I thought should always contain the username and password) into [`Server`](https://github.com/dimagi/couchdbkit/blob/bde99d120d855e201c069f0746dc3cc11f0886f9/couchdbkit/client.py#L79) where they are parsed and removed from the URL. Maybe this `db` is created by `<Server>.get_db(dbname)`, although I'm not sure where that happens...

Local testing confirmed that this new method of using `db._request_session` does indeed make a properly authenticated request.

Unfortunately our travis setup uses couch in admin party mode, so it does not test couch auth. It would be good to change it so the couch docker container requires authentication, but this is apparently non-trivial. I tried it locally, and was able to get my dockerized couch out of admin-party mode by removing `--with-admin-party-please` from the docker compose files and rebuilding the container. However, it was still allowing anonymous reads, and [fixing that may require creating a custom couch docker image](https://issues.apache.org/jira/browse/COUCHDB-3100).